private/pkg/git/gitobject/gitobject.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitobject
+
+import (
+	"time"
+
+	"github.com/bufbuild/buf/private/pkg/command"
+)
+
+// ID represents the ID of a Git object (tree, blob, or commit).
+type ID interface {
+	// Hex is the hexadecimal representation of this ID.
+	Hex() string
+	// String returns the hexadecimal representation of this ID.
+	String() string
+}
+
+// NewIDFromHex creates a new ID that is validated.
+func NewIDFromHex(value string) (ID, error) {
+	return parseObjectIDFromHex(value)
+}
+
+// Ident is a git user identifier. These typically represent authors and committers.
+type Ident interface {
```

Comment:
Is `ID` vs `Ident` a known git construct? If it's something from us, `ID` and `Ident` both are short for "Identity/Identifier"

Reply:
Identity is a known concept (see "Your Identity" section [here](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)), `ID` is our own, I could call it `Hash` instead, to better represent what it is (commit hash, tree hash, blob hash, etc.). What do you think?

Reply:
yea that makes sense